sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -594,16 +594,13 @@ qualifiedName
     : identifier ('.' identifier)*
     ;
 
-// Identifier that also allows the use of a number of SQL keywords (mainly for backwards compatibility).
-looseIdentifier
-    : identifier
-    | FROM
-    | TO
-    | TABLE
-    | WITH
+identifier
+    : strictIdentifier
+    | ANTI | FULL | INNER | LEFT | SEMI | RIGHT | NATURAL | UNION | INTERSECT | EXCEPT | DATABASE | ON | JOIN | CROSS
+    | SELECT | FROM | WHERE | HAVING | FROM | TO | TABLE | WITH | NOT
```

Comment:
Looks like we have to figure these ambiguities out case by case, which is kind of dangerous. How many other databases allow keyword as identifier?